tokenizers/src/pre_tokenizers/metaspace.rs

```diff
@@ -1,6 +1,45 @@
+use crate::tokenizer::{Decoder, PreTokenizedString, PreTokenizer, Result, SplitDelimiterBehavior};
 use serde::{Deserialize, Deserializer, Serialize};
+use std::fmt;
 
-use crate::tokenizer::{Decoder, PreTokenizedString, PreTokenizer, Result, SplitDelimiterBehavior};
+/// Enum representing options for the metaspace prepending scheme.
+#[derive(Debug, Clone, PartialEq, Serialize, Eq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum PrependScheme {
+    /// Specifies that the scheme should be prepended only once, on the first split.
+    First,
+    /// Specifies that the space should not be prepended.
+    Never,
+    /// Specifies that the scheme should always be prepended.
+    Always,
+}
+
+impl From<&str> for PrependScheme {
```

Comment:
This probably only belongs in the PyO3 layer. In pure rust the String doesn't seem very necessary.